scipy/stats/_discrete_distns.py

```diff
@@ -514,6 +526,142 @@ def _logcdf(self, k, M, n, N):
+
+    Consider a box containing :math:`N` balls:, :math:`K` red and
+    :math:`N-K` blue. We randomly sample balls from the box, one
+    at a time and *without* replacement, until we have picked :math:`r`
+    blue balls. `nhypergeom` is the distribution of the number of
+    red balls :math:`k` we have picked.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The symbols used to denote the shape parameters (`N`, `K`, and `r`) are not
+    universally accepted. See the Examples for a clarification of the
+    definitions used here.
+
+    The probability mass function is defined as,
+
+    .. math:: f(k; N, K, r) = \frac{{{k+r-1}\choose{k}}{{N-r-k}\choose{K-k}}}
+                                   {{N \choose K}}
+
+    for :math:`k \in [0, K]` and the binomial coefficient is:
+
+    .. math:: \binom{n}{k} \equiv \frac{n!}{k! (n - k)!}.
+
+    It is equivalent to observing :math:`k` successes in :math:`k+r-1`
+    samples with :math:`k+r`'th sample being a failure. The former
+    can be modelled as a hypergeometric distribution. The probability
+    of the latter is simply the number of failures remaining
+    :math:`N-K-(r-1)` divided by the size of the remaining population
+    :math:`N-(k+r-1)`. This relationship can be shown as:
+
+    .. math:: NHG(k;N,K,r) = HG(k;N,K,k+r-1)\frac{(N-K-(r-1))}{(N-(k+r-1))}
+
+    where :math:`NHG` is negative hypergeometric distribution and
+    :math:`HG` is the hypergeometric distribution.
+
+    %(after_notes)s
+
+    Examples
+    --------
+    >>> from scipy.stats import nhypergeom
+    >>> import matplotlib.pyplot as plt
+
+    Suppose we have a collection of 20 animals, of which 7 are dogs.
+    Then if we want to know the probability of finding a given number
+    of dogs (successes) in a sample with exactly 12 animals that
+    aren't dogs (failures), we can initialize a frozen distribution
+    and plot the probability mass function:
+
+    >>> N, K, r = [20, 7, 12]
+    >>> rv = nhypergeom(N, K, r)
+    >>> x = np.arange(0, K+2)
+    >>> pmf_dogs = rv.pmf(x)
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> ax.plot(x, pmf_dogs, 'bo')
+    >>> ax.vlines(x, 0, pmf_dogs, lw=2)
+    >>> ax.set_xlabel('# of dogs in our group with given 12 failures')
+    >>> ax.set_ylabel('nhypergeom PMF')
+    >>> plt.show()
+
+    Instead of using a frozen distribution we can also use `nhypergeom`
+    methods directly.  To for example obtain the probability mass
+    function, use:
+
+    >>> prb = nhypergeom.pmf(x, N, K, r)
+
+    And to generate random numbers:
+
+    >>> R = nhypergeom.rvs(N, K, r, size=10)
+
+    To verify the relationship between `hypergeom` and `nhypergeom`, use:
+
+    >>> from scipy.stats import hypergeom, nhypergeom
+    >>> HG = hypergeom._pmf
+    >>> NHG = nhypergeom._pmf
+    >>> N, K, r = 45, 13, 8
+    >>> k = 6
+    >>> NHG(k, N, K, r)
+    0.06180776620271643
+    >>> HG(k, N, K, k+r-1) * (N - K - (r-1)) / (N - (k+r-1))
+    0.06180776620271644
```

Comment:
Defining `HG` and `NHG` is more of a distraction than a help.  You are demonstrating the methods, so use the normal syntax for a method.
```suggestion
    >>> N, K, r = 45, 13, 8
    >>> k = 6
    >>> nhypergeom.pmf(k, N, K, r)
    0.06180776620271643
    >>> hypergeom.pmf(k, N, K, k+r-1) * (N - K - (r-1)) / (N - (k+r-1))
    0.06180776620271644
```

Reply:
done!